spec/unit/defaults_spec.rb

```diff
@@ -190,78 +190,29 @@
     end
   end
 
-  describe "the call hook for the cadir setting", unless: Puppet::Util::Platform.windows? do
-    it 'does not warn when the cadir is outside the puppet ssldir' do
-      FileUtils.mkdir_p(Puppet[:confdir])
-      # This is fun; in order to get the confdir setting from the spec_helper to
-      # not be deleted by the #unsafe_clear in Puppet::Settings#parse_config,
-      # we can code it into the confdir itself, and thus we can avoid getting
-      # that setting wiped out during Puppet.initialize_settings.
-      File.write(File.join(Puppet[:confdir], 'puppet.conf'),
-                 "cadir = /my/cool/path/for/my/cadir\n
-                  confdir = #{Puppet[:confdir]}")
-      expect(Puppet).to_not receive(:log_ca_migration_warning)
+  describe "the default cadir", :unless => Puppet::Util::Platform.windows?  do
+    it 'defaults to the puppetserver confdir when no cadir is found' do
       Puppet.initialize_settings
-      expect(Puppet[:cadir]).to eq("/my/cool/path/for/my/cadir")
+      expect(Puppet[:cadir]).to eq('/etc/puppetlabs/puppetserver/ca')
     end
 
-    it 'does warn when the cadir is inside the puppet ssldir' do
+    it 'defaults to the puppet/ssl dir when it detects a ca directory there' do
       FileUtils.mkdir_p(Puppet[:confdir])
-      cadir_location = File.join(Puppet[:ssldir], 'still_inside_ssldir')
+      cadir_location = File.join(Puppet[:ssldir], 'ca')
       # This is fun; in order to get the confdir setting from the spec_helper to
       # not be deleted by the #unsafe_clear in Puppet::Settings#parse_config,
       # we can code it into the confdir itself, and thus we can avoid getting
       # that setting wiped out during Puppet.initialize_settings.
       File.write(File.join(Puppet[:confdir], 'puppet.conf'),
                  "cadir = #{cadir_location}\n
```

Comment:
oh right.. I'm unclear how this a useful test of the default if we end up configuring the `cadir` here anyway..

Reply:
Ok, I'm just gonna go back to the old layout, which is probably more correct; you are right, this doesn't really work for this test.